addons/mail/models/res_users.py

```diff
@@ -35,13 +35,21 @@ class Users(models.Model):
         "Only internal user can receive notifications in Odoo",
     )]
 
-    @api.depends('share')
+    @api.depends('groups_id')
     def _compute_notification_type(self):
         for user in self:
-            # Only the internal users can receive notifications in Odoo
-            if user.share or not user.notification_type:
+            if user.has_group('mail.group_mail_notification_type_inbox'):
+                user.notification_type = 'inbox'
+            else:
                 user.notification_type = 'email'
 
+    def _inverse_notification_type(self):
+        for user in self:
+            if user.notification_type == 'inbox':
+                user.groups_id += self.env.ref('mail.group_mail_notification_type_inbox')
+            else:
+                user.groups_id -= self.env.ref('mail.group_mail_notification_type_inbox')
```

Comment:
We could store the `mail.group_mail_notification_type_inbox` before doing the for loop to avoid doing the `record.exists()` (done in the `ref` method) for each user in the loop. 🙂
```suggestion
        notification_type_group = self.env.ref('mail.group_mail_notification_type_inbox')
        for user in self:
            if user.notification_type == 'inbox':
                user.groups_id += notification_type_group
            else:
                user.groups_id -= notification_type_group
```

Reply:
Yup indeed, I am updating the code.